lib/logger/logging_util.js

```diff
@@ -1,8 +1,48 @@
 const Util = require('../util');
 
+const PROVIDED_TEXT = 'provided';
+const NOT_PROVIDED_TEXT = 'not provided';
+
 exports.describePresence = function (valueToDescribe) {
-  const VAR_PROVIDED_TEXT = 'provided';
-  const VAR_NOT_PROVIDED_TEXT = 'not provided';
+  return Util.isNotEmptyAsString(valueToDescribe) ? PROVIDED_TEXT : NOT_PROVIDED_TEXT;
+};
+
+/**
+ * @param {Object} sourceObject - The object holding attribute values.
+ * @param {Array<string>} attributesWithValues - Attributes to show with their values.
+ * @param {Array<string>} attributesWithoutValues - Attributes to show as present/not present.
+ * @returns {string} Comma-separated string describing the attributes.
+ */
+exports.attributesToString = function (
```

Comment:
i expect this to be quite a common case, that we describe a list of attributes from some object. It is not always enclosed with square brackets, since we may be merging multiple objects into 1 description.